// api/mockProducts.js
export default [
    {
      id: "1",
      categoryId: "1",
      name: "TIFFANY & CO.",
      description: "tiffany & co. edp 75ml mother day settiffany & co. edp 75ml mother day settiffany & co. edp 75ml mother day set",
      price: 235,
      image: "/uploads/TiffanyAndCo.png",
      size: "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
      weight: "132 grams (4.7 ounces)"
    },
    {
      id: "2",
      categoryId: "1",
      name: "ELIZABETH ARDEN",
      description: "Top Notes include Lily of the Valley Freesia Wild Violets. Mid Notes include Red Rose Moroccan Orange Flower Jasmine Ylang. Base Notes include Oakmoss Sandalwood Honey. An elegant floral bouquet of deep rich florals. A warm full inviting blend of sensuous notes. It's captivating sophisticated scent fills the senses and creates a signature for the woman who wears it.",
      price: 69,
      image: "/uploads/ELIZABETH ARDEN RED DOOR EDT.png",
      size: "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
      weight: "132 grams (4.7 ounces)"
    },
    {
      id: "3",
      categoryId: "4",
      name: "SWISSE ULTIBOOST",
      description: "Great tasting, premium quality formula to support collagen production and provide nutrients essential for healthy hair, skin and nails.",
      price: 35,
      image: "/uploads/Swisse Hair Skin Nails.png",
      size: "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
      weight: "132 grams (4.7 ounces)",
    },
    {
      id: "4",
      categoryId: "5",
      name: "GOAT MOISTURISE",
      description: "goat moisturising shampoo original 300mlgoat moisturising shampoo original 300mlgoat moisturising shampoo original 300mlgoat",
      price: 12,
      image: "/uploads/GOAT MOISTURISING SHAMPOO ORIGINAL 300ML.png",
      size: "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
      weight: "132 grams (4.7 ounces)"
    },
    {
      id: "5",
      categoryId: "3",
      name: "BIOGLAN KIDS",
      description: "Helps to support healthy growth and development the maintenance of healthy eyesight bones skin gums teeth and general health and wellbeing.",
      price: 19,
      image: "/uploads/BIOGLAN KIDS SMART VITA GUMMIES MULTI VITAMIN + VEGES 60S.png",
      size: "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
      weight: "132 grams (4.7 ounces)"
    },
    {
      id: "6",
      categoryId: "2",
      name: "ESTEE LAUDER",
      description: "Colour with the power to transform your lips your look your attitude. Intensifies: Saturates lips with high-intensity colour—in one stroke.",
      price: 70,
      image: "/uploads/ESTEE LAUDER PURE COLOR ENVY SCULPTING LIPSTICK.png",
      size: "124.4mm x 59.2mm x 8.97mm (4.9 x 2.33 x 0.35)",
      weight: "132 grams (4.7 ounces)"
    }
  ]
